quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/WildcardPropertyMapper.java

```diff
@@ -119,7 +123,10 @@ public boolean matchesWildcardOptionName(String name) {
 
     public Optional<String> getKcKeyForEnvKey(String envKey, String transformedKey) {
         if (transformedKey.startsWith(fromPrefix)) {
-            return Optional.ofNullable(getFrom(envKey.substring(fromPrefix.length()).toLowerCase().replace("_", ".")));
+            if (replacementChar != null) {
+                return Optional.ofNullable(getFrom(envKey.substring(fromPrefix.length()).toLowerCase().replace('_', replacementChar)));
+            }
+            return Optional.of(transformedKey);
```

Comment:
ATM, there's no way the replacementChar might be null, right? 

I'd rather throw an exception if the replacementChar is null and not returning the transformedKey.

Or what's the intention here?